File: selection.py
def selection_sort(arr: list) -> list:
    for i in range(len(arr)) :
        min_value = arr[i]
        min_index = i
        
        for j in range(len(arr)):
            if arr[j] > min_value:
                min_value = arr[j]
                min_index = j
                
                arr[i], arr[j] = arr[j], arr[i]
                
    return arr
        
        
        
        
arr = [5, 3, 6, 2, 1, 8]
res = selection_sort(arr)
print(res)

File: selection.js
function selection_sort(arr) {
    let minValue;
    let minIndex;

    for (i = 0; i<=arr.length - 1; i++) {
        minValue = arr[i]
        minIndex = i

        for (let j = 0; j < arr.length - 1; j++) {
            if (arr[j] > minValue) {
                minValue = arr[j]
                minIndex = j

                arr[j] = arr[i]
                arr[i] = minValue
            }
        }
    }

    return arr
}

arr = [6 ,7, 3, 2, 5,4, 1]
res = selection_sort(arr)
console.log(res)